### PR TITLE
Changed 'UpgradeOverlay' to 'WithColoredOverlay'

### LINKED ISSUE #12656: UpgradeOverlay trait needs to be renamed
I suggest `WithColoredOverlay`.

### PR DESCRIPTION
**Pull request by abc013**
Closes #12656.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Modifiers/WithColoredOverlay.cs [hunk 1]
```diff
@@ -16,17 +16,17 @@
 namespace OpenRA.Mods.Common.Traits
 {
 	[Desc("Display a colored overlay when a timed upgrade is active.")]
```
Comment (GraionDilach) [inline OpenRA.Mods.Common/Traits/Modifiers/WithColoredOverlay.cs]: still upgrade.

**Comment by abc013** [comment]
Updated.

**Review by GraionDilach — APPROVED** [review]
:+1:

File: OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs [hunk 2]
```diff
@@ -776,6 +776,11 @@ internal static void UpgradeActorRules(ModData modData, int engineVersion, ref L
 					}
 				}
 
+				// Rename UpgradeOverlay to WithColoredOverlay
+				if (engineVersion < 20170201)
+					if (node.Key.StartsWith("UpgradeOverlay", StringComparison.Ordinal))
+						node.Key = "WithColoredOverlay" + node.Key.Substring(14);
```
Comment (pchote) [inline OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs]: Please use `RenameNodeKey` instead of doing this manually.

**Review by pchote — DISMISSED** [review]

**Comment by abc013** [comment]
Again updated.

**Review by pchote — APPROVED** [review]

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/120a5ab5765b8b4d1823b18ed429d023a2aac2fc

**Pull request merged**